### DISCUSSION TITLE
OpenVINO iGPU and machine-learning hardware acceleration

### DISCUSSION THREAD

**Discussion by Eirikr70** [immich-app/immich#9105] [Q&A] [answered]
Hello,
I have just acquired an Odroid H4+, that has an iGPU compatible with OpenVINO. The documentation specifies "The server must have a discrete GPU, i.e. Iris Xe or Arc. Expect issues when attempting to use integrated graphics." Will this situation change in a foreseeable future ?
Thanks

**Comment by mertalev** [comment, marked as answer]
The requirement is due to an upstream issue, see #8226. It may change at some point. Feel free to try it anyway, just be aware that one of face detection or smart search have a good chance of failing.

**Reply by Eirikr70** [reply]
Thanks for that quick answer. I'll wait gently until it's solved, with hope !

**Reply by duskmoss** [reply]
Does 8826 being closed a few weeks ago now mean you expect intel igpu's to function for openvino?

**Reply by mertalev** [reply]
It should generally work now, yes.

**Reply by duskmoss** [reply]
Seems to be working well :) Very glad it got to a good state before the bigger slower model were added.

**Reply by conradkun** [reply]
Great to see this is working, it makes the decision of which CPU to buy much easier now that I'm building a NAS :) 

Should [the documentation](https://immich.app/docs/features/ml-hardware-acceleration/) be changed? It still lists only integrated Intel GPUs.

**Comment by Eirikr70** [comment]
Hello, OpenVino user here. I understand that in order to benefit from hardware acceleration, I have to modify the compose file in order to indicate the hardware to use. Should I also delete the model-cache volume ?

**Reply by mertalev** [reply]
You should, assuming you didn't set up immich in the last few weeks.
